client/cli/src/lib.rs

```diff
@@ -281,6 +288,21 @@ pub fn init(shared_params: &SharedParams, version: &VersionInfo) -> error::Resul
 	fdlimit::raise_fd_limit();
 	init_logger(shared_params.log.as_ref().map(|v| v.as_ref()).unwrap_or(""));
 
+	load_spec(config, shared_params, spec_factory)?;
+
+	if config.config_dir.is_none() {
+		config.config_dir = Some(base_path(&shared_params, version));
+	}
+
+	if config.database.is_none() {
+		config.database = Some(DatabaseConfig::Path {
+			path: config
+				.in_chain_config_dir(DEFAULT_DB_CONFIG_PATH)
+				.expect("We provided a base_path/config_dir."),
+			cache_size: None,
+		});
+	}
+
```

Comment:
This is breaking my changes. My motivation is to remove all the configuration setup from the `init()` method.

`init()` must be called only once. In cumulus we have multiple `Configuration` and they all need to be properly set up. This is not possible if you put config setup logic in this `init_method` because we won't want to call a second time `init` which will call `init_logger` again.

You need to create a private method `fill_config_dir_and_database_path` that will be called in `update_config_for_running_node` and all the `run()` commands that needs it in `client/cli/src/lib.rs`.

Reply:
I have done it in a similar way. But that just shows us the fucked up design of the CLI.